src/plugins/intel_cpu/src/nodes/executors/reduce.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "cpu_memory.h"
+#include "onednn/iml_type_mapper.h"
+#include "dnnl_scratch_pad.h"
+#include "executor.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+struct ReduceAttrs {
+    std::string nodeName;
+    std::vector<int> axes;
+    Algorithm operation;
+    bool keepDims;
+};
+
+class ReduceExecutor {
+public:
+    ReduceExecutor(const ExecutorContext::CPtr context);
+    virtual bool init(const ReduceAttrs& reduceAttrs,
+                      const std::vector<MemoryDescPtr>& srcDescs,
+                      const std::vector<MemoryDescPtr>& dstDescs,
+                      const dnnl::primitive_attr &attr) = 0;
+
+    virtual void exec(const std::vector<MemoryCPtr>& src, const std::vector<MemoryPtr>& dst, std::unordered_map<int, MemoryPtr> postOpsArgs) = 0;
+    virtual ~ReduceExecutor() = default;
+
+    virtual impl_desc_type getImplType() const = 0;
+
+    void setEngine(const dnnl::engine& engine) {
```

Comment:
There setters are not required anymore due to ExecutorContext introduction.
Same about corresponding class fields

Reply:
Removed.